saleor/checkout/test_checkout.py

```diff
@@ -164,3 +170,755 @@ def test_index_view(cart, status_code, url, rf):
     response = views.index_view(request, checkout)
     assert response.status_code == status_code
     assert response.url == url
+
+
+empty_cart = Mock(__len__=Mock(return_value=0))
+cart_with_shipping = Mock(__len__=Mock(return_value=1),
+                          is_shipping_required=Mock(return_value=True))
+cart_without_shipping = Mock(__len__=Mock(return_value=1),
+                             is_shipping_required=Mock(return_value=False))
+
+
+@pytest.fixture
+def shipping_method(db):
+    shipping_method = ShippingMethod.objects.create(name='Shipping method')
+    shipping_method_country = ShippingMethodCountry.objects.create(
+        country_code='PL', shipping_method=shipping_method,
+        price=Price('3.0', currency=settings.DEFAULT_CURRENCY))
+    return shipping_method_country
+
+
+@pytest.fixture
+def client_with_cart(client, product_in_stock):
+    product_variant = ProductVariant.objects.get()
+    url = product_variant.product.get_absolute_url()
+    data = {
+        'quantity': 2, 'variant': product_variant.id
+    }
+    client.post(url, data=data)
+    return client
+
+
+@pytest.fixture
+def without_shipping(monkeypatch):
+    monkeypatch.setattr(
+        'saleor.product.models.ProductVariant.is_shipping_required',
+        Mock(return_value=False))
+
+
+ADDRESS_DATA = {
+        'first_name': 'First Name', 'last_name': 'Last Name',
+        'company_name': 'Company Name', 'street_address_1': 'Street Address 1',
+        'street_address_2': 'Street Address2', 'city': 'City',
+        'postal_code': '54-321', 'country': 'PL'}
+
+
+DIFFERENT_ADDRES_DATA = {
+        'first_name': 'First Name', 'last_name': 'Last Name',
+        'company_name': 'Company Name', 'street_address_1': 'Another Street',
+        'city': 'Different City', 'postal_code': '54-321', 'country': 'PL'}
+
+
+@pytest.mark.parametrize(
+    'shipping_address_data, billing_address_data, update_billing_address_with', [
+        (ADDRESS_DATA, DIFFERENT_ADDRES_DATA, {'address': 'new_address'}),
+        (ADDRESS_DATA, {}, {'address': 'shipping_address'})
+    ])
+def test_checkout_whole_with_shipping_and_copying_address(
+        client_with_cart, shipping_method, shipping_address_data,
+        billing_address_data, update_billing_address_with):
+    """
+    Anonymous user with product in cart
+    Provides valid shipping address and method
+    Chooses shipping address as billing or provides new one
+    Successful order creates
```

Comment:
This looks more like a comment than a docstring. It does not really explain what the test does.